### DISCUSSION TITLE
Find Tenant Domain via Email

### DISCUSSION THREAD

**Discussion by dominikgeimer** [spatie/laravel-multitenancy#378] [Q&A]
Hello,

I really like this package, and it works without any problem. I am using the multi database approach. 

I want to find an email address inside a tenant. Furthermore, I think this is useful when a client forget its own subdomain.

Maybe someone can give me a hint. Possibly something like this: https://spatie.be/docs/laravel-multitenancy/v2/advanced-usage/looping-over-a-collection-of-tenants? But how would you guys do it?

Thanks!

**Comment by masterix21** [comment]
Take a look at this killer solution:

https://medium.com/swlh/querying-multiple-mysql-databases-at-once-7e92ae99f842

But if you are a dreamer like me, use meilisearch or another dedicated search engine.

**Reply by dominikgeimer** [reply]
Thanks for your reply. I think I found a solution: 

Just my simple test code, but it works. 
What do you think about it?
```
Route::get('/test', function () {
    $tenant = Tenant::all()->filterCurrent(function () {
        return User::where('email', 'jon.doe@acme.com')->first();
    });

    return $tenant->first()->domain;
});
```

**Reply by masterix21** [reply]
What happens when you will have 1.000 tenants (or more)? What happens if an email is in two tenants (or more)?

That's why I'm talking about a dedicated search engine. 

https://laravel.com/docs/9.x/scout
https://laravel-news.com/getting-started-laravel-scout-meilisearch
